### PR TITLE
3.x: Enable tests that were disabled during renaming to jakarta packages work

### PR DESCRIPTION
**Pull request by tvallin**
### Description

fix https://github.com/helidon-io/helidon/issues/7503

### Documentation

no impact

### CODE REVIEW

File: openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java [hunk 1]
```diff
@@ -561,8 +561,7 @@ private Optional<MediaType> chooseResponseMediaType(ServerRequest req) {
     private static class HelidonAnnotationScannerExtension implements AnnotationScannerExtension {
 
         @Override
-        public Object parseExtension(String key, String value) {
-
+        public Object parseValue(String value) {
```
Comment (tjquinno) [inline openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java:564]: I do not see why we need to create `parseValue` and use it from `parseExtension`. IIRC `parseValue` is a new method added to the SmallRye library in a later release (we use the later release in Helidon 4.x but not in 3.x). I notice that the new `parseValue` method here does not have `@Override` as it does in 4.x (which makes sense given that the method does not exist yet in SmallRye's library, at least from the Helidon 3.x perspective).

That said, the change seems to do no harm so I'm OK with it. If I have missed something about why it was added please let me know.

File: examples/integrations/micronaut/data/src/main/java/io/helidon/examples/integrations/micronaut/data/BeanValidationExceptionMapper.java [hunk 2]
```diff
@@ -16,7 +16,8 @@
 
 package io.helidon.examples.integrations.micronaut.data;
 
-import jakarta.validation.ConstraintViolationException;
+import javax.validation.ConstraintViolationException;
```
Comment (tjquinno) [inline examples/integrations/micronaut/data/src/main/java/io/helidon/examples/integrations/micronaut/data/BeanValidationExceptionMapper.java:19]: Should this actually be changed from `jakarta` to `javax`? 

Same question for the similar change in `PetResource.java` below.

File: openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java (hunk 1 above)
Comment (tvallin) [reply openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java:564]: When enabling test on `openapi-basic` example. It throws a class def not found exception about `jackson`.
In 3.x, the smallrye library version used is `2.1.16` and the `AnnotationScannerExtension` uses the `parseValue` with default implementation and uses `jackson`.
The new `parseValue` from my PR does have the `@Override` annotation ... line 563.
You can easily reproduce the issue by running `openapi-basic` example and enabling the test.

File: examples/integrations/micronaut/data/src/main/java/io/helidon/examples/integrations/micronaut/data/BeanValidationExceptionMapper.java (hunk 2 above)
Comment (tvallin) [reply examples/integrations/micronaut/data/src/main/java/io/helidon/examples/integrations/micronaut/data/BeanValidationExceptionMapper.java:19]: With the current version of `micronaut`, it does not handle `jakarta.validation`.

Here is the PR : 
Revert back to using javax.validation as jakarta.validation is not supported by micronaut [#4476 ](https://github.com/helidon-io/helidon/pull/4476)

**Review by tjquinno — APPROVED** [review]

**Pull request merged**